keel-artifact/src/test/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactDeployingListenerTests.kt

```diff
@@ -0,0 +1,66 @@
+package com.netflix.spinnaker.keel.artifact
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.id
+import com.netflix.spinnaker.keel.events.ArtifactVersionDeploying
+import com.netflix.spinnaker.keel.test.DummyResourceSpec
+import com.netflix.spinnaker.keel.test.combinedMockRepository
+import com.netflix.spinnaker.keel.test.deliveryConfig
+import com.netflix.spinnaker.keel.test.resource
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+
+internal class ArtifactDeployingListenerTests : JUnit5Minutests {
+  class Fixture {
+    val realResource: Resource<DummyResourceSpec> = resource()
+    val r: Resource<DummyResourceSpec> = mockk()
+    val config = deliveryConfig(resource = r)
+    val artifact = config.artifacts.first()
+    val repository = combinedMockRepository()
+
+    val event = ArtifactVersionDeploying(
+      resourceId = realResource.id,
+      artifactVersion = "1.1.1"
+    )
+
+    val subject = ArtifactDeployingListener(repository)
+  }
+
+  fun tests() = rootContext<Fixture> {
+    fixture { Fixture() }
+
+    before {
+      every { r.id } returns realResource.id
+      every { r.metadata } returns realResource.metadata
+      every { repository.getResource(r.id) } returns realResource
+      every { repository.deliveryConfigFor(r.id) } returns config
+      every { repository.environmentFor(r.id) } returns config.environments.first()
+    }
+
+    context("no artifact") {
+      before {
+        every { r.findAssociatedArtifact(config) } returns null
+      }
+
+      test("nothing is marked deploying") {
+        subject.onArtifactVersionDeploying(event)
+        verify(exactly = 0) { repository.isApprovedFor(config, any(), event.artifactVersion, any()) }
```

Comment:
I recently learned from @robfletcher that you can write this in a much more human-friendly way:
```kotlin
verify { repository.isApprovedFor(config, any(), event.artifactVersion, any()) wasNot Called }
```